cypress/e2e/flow-builder/blocks/moving.cy.ts

```diff
@@ -0,0 +1,74 @@
+import {IBlock} from '@floip/flow-runner'
+import Chainable = Cypress.Chainable
+
+describe('Moving a block', () => {
+  beforeEach(() => {
+    cy.createFlow({
+      label: 'Moving a block',
+    })
+
+    cy.addBlock(['Content', 'Message']).then(blockId => {
+      cy.get(`[data-cy="block--${blockId}"]`)
+        .find('[data-cy^="block-handle--"]')
+        .as('blockArea')
+    })
+
+    // wait for flow validation to succeed and flow validation toolbar to disappear
+    // to avoid interference with block coordinates
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(200)
+  })
+
+  it('moves the block', () => {
+    const deltaX = 200
+    const deltaY = 300
+
+    // eslint-disable-next-line arrow-body-style
+    return getBlockVisualCoordinates().then(({x: oldVisualX, y: oldVisualY}) => {
+      return getBlockMetadataCoordinates().then(({x: oldMetadataX, y: oldMetadataY}) => {
+        cy.get('@blockArea').move({deltaX, deltaY})
+
+        // eslint-disable-next-line arrow-body-style
+        return getBlockVisualCoordinates().then(({x: newVisualX, y: newVisualY}) => {
+          return getBlockMetadataCoordinates().then(({x: newMetadataX, y: newMetadataY}) => {
+            cy.log('old visual x,y:', oldVisualX, oldVisualY)
+            cy.log('new visual x,y:', newVisualX, newVisualY)
+
+            cy.wrap(newVisualX - oldVisualX).should('eq', deltaX)
+            cy.wrap(newVisualY - oldVisualY).should('eq', deltaY)
+
+            cy.log('old metadata x,y:', oldMetadataX, oldMetadataY)
+            cy.log('new metadata x,y:', newMetadataX, newMetadataY)
+
+            cy.wrap(newMetadataX - oldMetadataX).should('eq', deltaX)
+            cy.wrap(newMetadataY - oldMetadataY).should('eq', deltaY)
+          })
+        })
+      })
+    })
+  })
+})
```

Comment:
I was able to extract the `getBlockVisualCoordinates` and `getBlockMetadataCoordinates` for reusability.
But mostly for readability (it was aweful before).

It's still nested and I'm happy with it.
But any ideas on how to improve it further are very welcome.

Reply:
I'm good with them for now